src/main/java/org/jitsi/protocol/xmpp/util/MediaSourceMap.java

```diff
@@ -236,33 +217,33 @@ public SourcePacketExtension findSourceViaSsrc(String media, long ssrc)
     public String getMediaTypeForSource(SourcePacketExtension source)
     {
         Set<String> mediaTypes = getMediaTypes();
-        for (String mediaType : mediaTypes)
-        {
-            if (findSource(mediaType, source) != null)
-                return mediaType;
-        }
-        return null;
+        return mediaTypes.stream()
+            .filter(mediaType -> findSource(mediaType, source) != null)
+            .findFirst()
+            .orElse(null);
     }
 
     /**
      * Finds SSRC for given owner.
      *
-     * @param media the media type of the SSRC we'll be looking for.
+     * @param mediaType the media type of the SSRC we'll be looking for.
      * @param owner the MUC JID  of the SSRC owner.
      *
      * @return <tt>SourcePacketExtension</tt> for given media type and owner or
-     *         <tt>null</tt> if not found.
+     *  <tt>null</tt> if not found.
      */
-    public SourcePacketExtension findSSRCforOwner(String media, Jid owner)
+    public SourcePacketExtension findSsrcForOwner(String mediaType, Jid owner)
     {
-        List<SourcePacketExtension> mediaSSRCs = getSourcesForMedia(media);
-        for (SourcePacketExtension ssrc : mediaSSRCs)
-        {
-            Jid ssrcOwner = SSRCSignaling.getSSRCOwner(ssrc);
-            if (ssrcOwner != null && ssrcOwner.equals(owner))
-                return ssrc;
-        }
-        return null;
+        List<SourcePacketExtension> mediaSSRCs = getSourcesForMedia(mediaType);
+        return mediaSSRCs.stream()
+            .filter(
+                ssrc ->
+                {
+                    Jid ssrcOwner = SSRCSignaling.getSSRCOwner(ssrc);
+                    return ssrcOwner != null && ssrcOwner.equals(owner);
+                })
+            .findFirst()
+            .orElse(null);
```

Comment:
could be:
```
return mediaSSRCs.stream()
    .map(ssrc -> SSRCSignaling.getSSRCOwner(ssrc))
    .filter(Objects::nonNull)
    .filter(ssrcOwner -> ssrcOwner.equals(owner))
    .findFirst()
    .orElse(null)
```

(i don't care if you change it, i just had to take a break going through the PR :)

Reply:
I tried, but the call to map() loses track of the object we want to return

Reply:
oh oops, didn't notice that